Sources/NIOFileSystem/FileChunks.swift

```diff
@@ -85,61 +85,89 @@ extension FileChunks.FileChunkIterator: Sendable {}
 // MARK: - Internal
 
 @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
-extension BufferedStream where Element == ByteBuffer {
+fileprivate typealias FileChunkSequenceProducer = NIOThrowingAsyncSequenceProducer<ByteBuffer, Error, NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark, FileChunkProducer>
+
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+extension NIOThrowingAsyncSequenceProducer where Element == ByteBuffer, Failure == Error,
+        Strategy == NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark, Delegate == FileChunkProducer {
     static func makeFileChunksStream(
         of: Element.Type = Element.self,
         handle: SystemFileHandle,
         chunkLength: Int64,
         range: FileChunks.ChunkRange,
         lowWatermark: Int,
         highWatermark: Int
-    ) -> BufferedStream<ByteBuffer> {
-        let state: ProducerState
-        switch range {
-        case .entireFile:
-            state = ProducerState(handle: handle, range: nil)
-        case .partial(let partialRange):
-            state = ProducerState(handle: handle, range: partialRange)
-        }
-        let protectedState = NIOLockedValueBox(state)
+    ) -> FileChunkSequenceProducer {
 
-        var (stream, source) = BufferedStream.makeStream(
-            of: ByteBuffer.self,
-            backPressureStrategy: .watermark(low: lowWatermark, high: highWatermark)
-        )
-
-        source.onTermination = {
-            protectedState.withLockedValue { state in
-                state.done()
-            }
-        }
-
-        // Start producing immediately.
         let producer = FileChunkProducer(
-            state: protectedState,
-            source: source,
-            path: handle.path,
+            range: range,
+            handle: handle,
             length: chunkLength
         )
-        producer.produceMore()
 
-        return stream
+        let nioThrowingAsyncSequence = NIOThrowingAsyncSequenceProducer.makeSequence(
+            elementType: ByteBuffer.self,
+            backPressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark(
+                lowWatermark: 4,
+                highWatermark: 8
+            ),
+            finishOnDeinit: false,
+            delegate: producer
+        )
+
+        producer.setSource(nioThrowingAsyncSequence.source)
+
+        return nioThrowingAsyncSequence.sequence
     }
 }
 
 @available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
-private struct FileChunkProducer: Sendable {
+private final class FileChunkProducer: NIOAsyncSequenceProducerDelegate, Sendable {
     let state: NIOLockedValueBox<ProducerState>
-    let source: BufferedStream<ByteBuffer>.Source
+
     let path: FilePath
     let length: Int64
 
+    init(range: FileChunks.ChunkRange, handle: SystemFileHandle, length: Int64) {
+
```

Comment:
nit: no blank lines at the start of funcs
```suggestion
    init(range: FileChunks.ChunkRange, handle: SystemFileHandle, length: Int64) {
```